js/src/Components/Widgets/LinkWidget.js

```diff
@@ -58,77 +59,196 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
+      ].join('')),
+      loose: new RegExp([
+        '(?:(?![^:@]+:[^:@\\/]*@)([^:\\/?#.]+):)?',
+        '(?:\\/\\/\\/?)?',
+        '((?:(([^:@\\/]*):?([^:@\\/]*))?@)?([^:\\/?#]*)(?::(\\d*))?)',
+        '(((\\/(?:[^?#](?![^?#\\/]*\\.[^?#\\/.]+(?:[?#]|$)))*\\/?)?([^?#\\/]*))',
+        '(?:\\?([^#]*))?(?:#(.*))?)'
+      ].join(''))
+    }
+
+    let m = parser[mode].exec(url)
+    let uri = {}
+    let i = 14
+
+    while (i--) {
+      if (m[i]) {
+        uri[key[i]] = m[i]
       }
-      return {
-        protocol: null,
-        pathname: decodeURIComponent(parser.pathname + parser.search + parser.hash)
-      };
     }
-    return {
-      protocol: parser.protocol,
-      pathname: parser.pathname + parser.search + parser.hash
-    };
-  };
 
-  const getUriFromString = (userString) => {
-    var uri = userString.trim();
-    if (uri.length === 0) {
-      return uri;
+    if (component) {
+      return uri[component.replace('PHP_URL_', '').toLowerCase()]
     }
-    const matchedEntity = userString.match(/.+\s\(([^\)]+)\)/);
-    const parsedUrl = parseUrl(userString);
 
-    if (matchedEntity && matchedEntity.length === 2) {
-      uri = `entity:node/${matchedEntity[1]}`
+    if (mode !== 'php') {
+      let name = 'queryKey'
+
+      parser = /(?:^|&)([^&=]*)=?([^&]*)/g
+      uri[name] = {}
+      query = uri[key[12]] || ''
+      query.replace(parser, function ($0, $1, $2) {
+        if ($1) {
+          uri[name][$1] = $2
+        }
+      })
+    }
+
+    delete uri.source
+    return uri
+  };
+
+  /**
+   * Similar to the link field widget function in PHP.
+   *
+   * @see \Drupal\link\Plugin\Field\FieldWidget\LinkWidget::getUserEnteredStringAsUri()
+   *
+   * @param string
+   * @returns {*}
+   */
+  const getUserEnteredStringAsUri = (string) => {
+    // By default, assume the entered string is an URI.
+    let uri = string.trim().toLowerCase();
+
+    // Detect entity autocomplete string, map to 'entity:' URI.
+    const entity_id = extractEntityIdFromAutocompleteInput(uri);
+    if (entity_id !== null) {
+      // @todo Support entity types other than 'node'. Will be fixed in
+      //    https://www.drupal.org/node/2423093.
+      uri = 'entity:node/' + entity_id;
+    }
+    // Support linking to nothing.
+    else if (['<nolink>', '<none>'].includes(string)) {
+      uri = 'route:' + string;
     }
-    else if (parsedUrl.protocol === null) {
-      userString = parsedUrl.pathname;
-      if (userString.trim().indexOf('<front>') === 0) {
-        userString = '/' + userString.substr(7);
+    // Detect a schemeless string, map to 'internal:' URI.
+    else if (string.length > 0 && parseUrl(string, 'PHP_URL_SCHEME') === undefined) {
+      // @todo '<front>' is valid input for BC reasons, may be removed by
+      //   https://www.drupal.org/node/2421941
+      // - '<front>' -> '/'
+      // - '<front>#foo' -> '/#foo'
+      if (string.indexOf('<front>') === 0) {
+        string = '/' + string.substr(7);
       }
```

Comment:
You're likely going to have to handle the `<none>` syntax as well.

The app crashes when I put in a `<none>`

```
TypeError: n.ltrim is not a function
```

Steps to reproduce:
1. From a fresh build of 1.x cardinalsites, Check out this branch
2. Create new basic page
3. Create a new card
4. Enter `<none>` for the link option and unfocus off the field.
5. Crash pants.

Reply:
Thanks for catching that. the Crash Pants is a no good. FWIW, the data is saved correctly without breaking, but the card twig renders out an `<a href="" class="su-link su-button">Some Text</a>` without any href. But that's a bug for another area.